test/test_gpu_device_detection.py

```diff
@@ -0,0 +1,34 @@
+import os
+import unittest
+
+import torch
+import torch.cuda
+import torch_xla
+import torch_xla.core.xla_env_vars as xenv
+import torch_xla.core.xla_model as xm
+import torch_xla.runtime as xr
+
+
+@unittest.skipIf(
+    os.getenv(xenv.PJRT_DEVICE) != None or not torch.cuda.is_available(),
```

Comment:
Wondering if you need to check `os.getenv(xenv.PJRT_DEVICE) != None` since you are doing
```
    os.unsetenv(xenv.PJRT_DEVICE)
    os.unsetenv(xenv.GPU_NUM_DEVICES)
```
As long as it's gpu ci, regardless if PJRT_DEVICE is set, you can run this test, right?

Reply:
No it doesn't run. The problem is setUp() runs after, so it would unset xenv.PJRT_DEVICE when it's explicitly set on CPU CI/CD. I think that CI still has CUDA, so then it runs the GPU tests. In a follow up, I can remove the PJRT_DEVICE=CUDA from CI/CD so that the tests run.